kernel/ramfs.c

```diff
@@ -2401,6 +2403,165 @@ static int _fs_futimens(
     return ret;
 }
 
+static int _chown(inode_t* inode, uid_t owner, gid_t group)
+{
+    int ret = 0;
+
+    if (owner != -1u)
+        inode->uid = owner;
+
+    if (group != -1u)
+        inode->gid = group;
+
+    /* For executables, clear set-user-ID and set-group-ID bits */
+    if (inode->mode & (S_IXUSR | S_IXGRP | S_IXOTH))
+    {
+        if (inode->mode & S_ISUID)
+            inode->mode &= ~S_ISUID;
+
+        /* Exclude clearing group id for non-group executables */
+        if ((inode->mode & S_ISGID) && (inode->mode & S_IXGRP))
+            inode->mode &= ~S_ISGID;
+    }
+
```

Comment:
Update modified timestamp?

Reply:
This is almost identical to the same function in ext2fs. I wonder if we can share utility functions between different types of file system implementations.

Reply:
As discussed offline, inode structs being different ext2fs and ramfs makes having shared functions difficult.